module-bsp/board/rt1051/puretx/hal/battery_charger/CurrentMeasurementScope.cpp

```diff
@@ -0,0 +1,39 @@
+// Copyright (c) 2017-2022, Mudita Sp. z.o.o. All rights reserved.
+// For licensing, see https://github.com/mudita/MuditaOS/LICENSE.md
+
+#include "CurrentMeasurementScope.hpp"
+
+extern "C"
+{
+#include "FreeRTOS.h"
+}
+#include <timers.h>
+#include <bsp/battery_charger/battery_charger.hpp>
+#include <ticks.hpp>
+
+namespace hal::battery::CurrentMeasurementScope
+{
+    namespace
+    {
+        constexpr auto samplingTime_MS = 100;
+        TimerHandle_t samplingTimerHandle;
+
+        static void getSample(TimerHandle_t xTimer)
+        {
+            LOG_DEBUG("[scope]: { \"timestamp\" : %ld, \"current\" : %d, \"current_filtered\" : %d  }",
+                      cpp_freertos::Ticks::TicksToMs(cpp_freertos::Ticks::GetTicks()),
+                      bsp::battery_charger::getCurrentMeasurement(),
+                      bsp::battery_charger::getAvgCurrent());
```

Comment:
the best solution would be some binary protocol instead of the JSON - it would save some data transfer.
What about the separate RTT channel?

Reply:
alright, just asking ;)